### PR TITLE
Shuffleboard abstractions

### PR DESCRIPTION
**Pull request by elliotnash**
abstractions for the SendableChooser class

### CODE REVIEW

File: src/main/java/frc/team997/lib/util/ShuffleboardChooser.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    /**
+     * Constructs a ShuffleboardChooser from an enum implementing {@link ShuffleboardOption}
+     *
+     * @param enumClass The {@code <enum>.class}
+     * @param defaultOption The default enum option
+     * @return A new ShuffleboardChooser object
+     * @param <T> The enum's type
+     */
+    @SuppressWarnings("unchecked")
+    public static <T> ShuffleboardChooser<T> fromEnum(
```
Comment (tandej) [inline src/main/java/frc/team997/lib/util/ShuffleboardChooser.java]: If we’re moving to glass for dashboards, it makes sense to give this a more generic name.

File: src/main/java/frc/team997/lib/util/ShuffleboardChooser.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+    /**
+     * Constructs a ShuffleboardChooser from a {@code Map<String, T>}
+     *
+     * @param options The option list, as a Map<title, value>
+     * @param defaultOption The title of the default option
+     */
+    public ShuffleboardChooser(Map<String, T> options, String defaultOption) {
+        // I watched this break the drivetrain - every shuffleboard option should
+        // have a default, but if not the default should be explicitly passed as null.
```
Comment (tandej) [inline src/main/java/frc/team997/lib/util/ShuffleboardChooser.java]: Not an issue— but it HAD a default, it just wasn’t working for whatever reason

File: src/main/java/frc/team997/lib/util/ShuffleboardChooser.java (hunk 1 above)
Comment (elliotnash) [reply src/main/java/frc/team997/lib/util/ShuffleboardChooser.java]: I agree - do you have any suggestions?

Comment (tandej) [reply src/main/java/frc/team997/lib/util/ShuffleboardChooser.java]: OptionSwitcher?
DashboardChooser?

Comment (elliotnash) [reply src/main/java/frc/team997/lib/util/ShuffleboardChooser.java]: combine the two - SendableDashboardOptionSwitcherChooser

let’s make a factory while we’re at it

but seriously, DashboardChooser sounds good i’ll make the change

**Comment by tandej** [comment]
Oh-- once the Logger rewrite gets pushed this should add an option (construction parameter defaulting to true?) of whether to log changes here

**Comment by elliotnash** [comment]
> Oh-- once the Logger rewrite gets pushed this should add an option (construction parameter defaulting to true?) of whether to log changes here

Can we merge this and can you include that in your logger PR? it can just go right before the call listeners loop in onOptionUpdate

**Review by tandej — APPROVED** [review]

**Pull request merged**